### PR TITLE
Implement the Code Changes for Upgrading the Owners API Endpoint

### PR DESCRIPTION
**Pull request by sgandhi1311**
# Description of change
This PR implements the necessary code changes to upgrade the Owners API endpoint (from v2 to v3) in the tap-Hubspot integration. The upgrade is essential to ensure compatibility with the latest version of the API and to enhance overall functionality.

# Manual QA steps
 - Performed manual testing to ensure the endpoint works as expected.
 
# Risks
 - Low, as it won't affect the existing connections as it is major version bump.
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_hubspot/schemas/owners.json [hunk 1]
```diff
@@ -1,72 +1,48 @@
...
-                    },
-                    "remoteType": {
-                        "type": ["null", "string"]
-                    },
-                    "active": {
-                        "type": ["null", "boolean"]
-                    }
-                }
+      "id": {
+        "type": ["null", "string"]
```
[the remark this answers is not in the document]
Comment (RushiT0122) [reply tap_hubspot/schemas/owners.json:5]: Confirmed its `string` now.

File: tap_hubspot/__init__.py [hunk 2]
```diff
@@ -850,7 +850,6 @@ def sync_tickets(STATE, ctx):
     singer.write_state(STATE)
     return STATE
 
-
 # NB> no suitable bookmark is available: https://developers.hubspot.com/docs/methods/email/get_campaigns_by_id
 def sync_campaigns(STATE, ctx):
     catalog = ctx.get_catalog_from_id(singer.get_currently_syncing(STATE))
```
Comment (RushiT0122) [inline tap_hubspot/__init__.py:855]:
```suggestion
    singer.write_state(STATE)
    return STATE


# NB> no suitable bookmark is available: https://developers.hubspot.com/docs/methods/email/get_campaigns_by_id
def sync_campaigns(STATE, ctx):
    catalog = ctx.get_catalog_from_id(singer.get_currently_syncing(STATE))
```

File: tap_hubspot/__init__.py [hunk 3]
```diff
@@ -1047,39 +1046,48 @@ def sync_workflows(STATE, ctx):
...
-    bookmark_key = 'updatedAt'
+    stream_id = "owners"
+    primary_key = "id"
+    bookmark_key = "updatedAt"
 
-    singer.write_schema("owners", schema, ["ownerId"], [bookmark_key], catalog.get('stream_alias'))
-    start = get_start(STATE, "owners", bookmark_key)
-    max_bk_value = start
+    max_bk_value = bookmark_value = utils.strptime_with_tz(
+        get_start(STATE, stream_id, bookmark_key))
```
Comment (RushiT0122) [inline tap_hubspot/__init__.py:1059]:
```suggestion
    bookmark_value = utils.strptime_with_tz(
        get_start(STATE, stream_id, bookmark_key))
    max_bk_value = bookmark_value
```

File: tap_hubspot/__init__.py [hunk 4]
```diff
@@ -1047,39 +1046,48 @@ def sync_workflows(STATE, ctx):
...
+    with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING) as transformer:
         # To handle records updated between start of the table sync and the end,
         # store the current sync start in the state and not move the bookmark past this value.
         sync_start_time = utils.now()
-        for row in data:
-            record = bumble_bee.transform(lift_properties_and_versions(row), schema, mdata)
-            if record[bookmark_key] >= max_bk_value:
-                max_bk_value = record[bookmark_key]
+        for row in gen_request_tickets(stream_id, url, params, 'results', "paging"):
+            # parsing the string formatted date to datetime object
```
Comment (RushiT0122) [inline tap_hubspot/__init__.py:1075]:
```suggestion
            # Parsing the string formatted date to datetime object
```

File: tap_hubspot/__init__.py [hunk 5]
```diff
@@ -1047,39 +1046,48 @@ def sync_workflows(STATE, ctx):
...
+            modified_time = utils.strptime_to_utc(row[bookmark_key])
 
-            if record[bookmark_key] >= start:
-                singer.write_record("owners", record, catalog.get('stream_alias'), time_extracted=time_extracted)
+            # Checking the bookmark value is present on the record and it
+            # is greater than or equal to defined previous bookmark value
+            if modified_time and modified_time >= bookmark_value:
+                # transforms the data and filters out the selected fields from the catalog
```
Comment (RushiT0122) [inline tap_hubspot/__init__.py:1081]:
```suggestion
                # Transforms the data and filters out the selected fields from the catalog
```

File: tap_hubspot/__init__.py [hunk 6]
```diff
@@ -1047,39 +1046,48 @@ def sync_workflows(STATE, ctx):
...
+                record = transformer.transform(lift_properties_and_versions(row), schema, mdata)
+                singer.write_record(stream_id, record, catalog.get(
+                    'stream_alias'), time_extracted=utils.now())
+            if modified_time and modified_time >= max_bk_value:
+                max_bk_value = modified_time
```
Comment (RushiT0122) [inline tap_hubspot/__init__.py:1086]:
```suggestion
            if modified_time and modified_time >= bookmark_value:
                # transforms the data and filters out the selected fields from the catalog
                record = transformer.transform(lift_properties_and_versions(row), schema, mdata)
                singer.write_record(stream_id, record, catalog.get(
                    'stream_alias'), time_extracted=utils.now())

                if modified_time >= max_bk_value:
                    max_bk_value = modified_time
```

**Review by RushiT0122 — CHANGES_REQUESTED** [review]
Requested few changes inline.

File: tap_hubspot/__init__.py (hunk 3 above)
Comment (sgandhi1311) [reply tap_hubspot/__init__.py]: done

File: tap_hubspot/__init__.py (hunk 6 above)
Comment (sgandhi1311) [reply tap_hubspot/__init__.py]: Ideally, modified_time should never be `None` as per the docs. And also `updatedAt` is the automatic field so it is never expected to be None/null.

**Review by RushiT0122 — APPROVED** [review]

**Pull request merged**